FS-28 update css override documentation

biz123 opened the pull request:
Completes [FS-28](https://palantir.atlassian.net/browse/FS-28)

To Test:
Review documentation and examples for clarity. 

Notes:
Existing documentation looked good so I only added some notes about updating existing override CSS should you be going from 2. to 3.x. 

CSS and SCSS files are updated to match namespace changes. 

Random note: I accidentally pushed to the 3.x branch so I revert-committed and  cherry-picked to here.

biz123 committed:
update override scss and css

biz123 committed:
updates CSS override documentation.

agentrickard commented on docs/assets/search_theme_override.css:
```diff
@@ -1,335 +1,265 @@
-  background-color: #f6f6f6;
-}
+aside.fs-aside {
+  background-color: #f6f6f6; }
```
Is this the new style standard? To have the closing brace on the same line?

biz123 replied:
That is curious. It was compiled using the toolset on the react app. I will look at how that was set up.

biz123 replied:
Since the intent here is really a template for someone not compiling SASS but editing CSS directly I should not have used compiled CSS. I will review/update this to more user-friendly and will revert those braces.

agentrickard reviewed (changes requested):
Looks fine except I don't understand why closing braces aren't on a new line anymore.

agentrickard commented:
I have a semi-related question.

When changing the CSS, are we now using `fs-root` as the base ID of the application? 

If so, that has cascading effects throughout the application.

biz123 committed:
Merge branch '8.x-3.x' into FS-28-update-CSS-override-documentation

biz123 commented:
> I have a semi-related question.
> 
> When changing the CSS, are we now using `fs-root` as the base ID of the application?
> 
> If so, that has cascading effects throughout the application.

I can address that here.

biz123 committed:
updates root ID to fs-root

biz123 committed:
CSS cleanup

agentrickard commented:
We now have to go into the actual React code and change that, too. I wasn't sure if you had done that when you updated the CSS.

See, for instance,  https://github.com/palantirnet/federated-search-react/blob/2.x/src/index.js#L163

biz123 committed:
run CSScomb

biz123 committed:
runs CSS comb on scss

biz123 commented:
@agentrickard I have updated the CSS and SASS with some code cleanup. I also updated the main app ID in the template in templates/search-app.html.twig and related documentation.

biz123 committed:
minor readme edits

biz123 commented:
I just reviewed this again to make sure no updates are needed after completing and merging https://github.com/palantirnet/federated-search-react/pull/62
...And this still works with latest updates so it is ready for review.

agentrickard commented:
I rolled a 3.0.0-beta1 version of the app and will include it here for testing.

agentrickard committed:
Use the 3.0.0-beta1 react libraries.

agentrickard commented:
When I test this in Drupal 8, I get a flash of unstyled content, which looks like the JS hasn't fully loaded.

This did not happen in the Alpha1 release, so I wonder if it's a CSS issue?

See https://drive.google.com/file/d/1UlcCjkrmhXwGkS5f7OBXvLAHz-44sfvM/view?usp=sharing

agentrickard commented:
Gif version

![Untitled](https://user-images.githubusercontent.com/360988/76328020-23612280-62c1-11ea-81ea-20e646d516e2.gif)

agentrickard commented:
I tried reverting https://github.com/palantirnet/federated-search-react/pull/62/commits/3d93f2acb2c6c5c852b419782e070fb2f8b20c43 and rolling a beta2 release from the `test-normalize` branch.

I still get a flash of the JS message.

agentrickard committed:
Try the beta2 version of the css

nstriedinger committed:
fix template for flicker

nstriedinger commented:
The issue:
![Screen Shot 2020-03-12 at 5 17 03 PM](https://user-images.githubusercontent.com/231467/76568082-864fe680-6486-11ea-9b96-e22ee4b9a2af.png)

agentrickard commented:
We need to apply these changes to the 7.x version, which I can do today.
